canonical/microcloud-cluster-manager
# feat(snap) add snap configuration WD-11369

# Done
- added snapcraft configuration
- fix the ui crashing when empty list of clusters is returned as `null` initially

# QA
1. Build the package locally: Install snapcraft with `snap install snapcraft`, checkout this branch, then run `snapcraft` from the top of this branch.
2. Install the locally built snap file, which should be in the repo directory with `sudo snap install --dangerous ./lxd-cluster-manager_0.1.0_amd64.snap`
3. Configure the service via terminal i.e. with
```
sudo lxd-cluster-manager init "member1" 0.0.0.0:9901 --bootstrap --control-address 0.0.0.0:9910
sudo lxd-cluster-manager config set oidc.issuer=https://lxd-ui-demo.us.auth0.com/
sudo lxd-cluster-manager config set oidc.client.id=OZSAeCbqAXZid3LL1gRQEkLXP9KlwZtJ
sudo lxd-cluster-manager config set oidc.audience=https://lxd-ui-demo.us.auth0.com/api/v2/
```
4. Now you can use the UI to register a cluster / connect a lxd remote cluster via https://127.0.0.1:9901/

Edit: Alternative is to `snap install lxd-cluster-manager --channel=latest/edge` to skip steps 1. and 2. above. I uploaded a local build to the snap store with the state of this branch.
